bitmap.go

```diff
@@ -0,0 +1,41 @@
+package render
+
+import (
+	"bytes"
+	"image"
+	"image/draw"
+	_ "image/jpeg" // load image formats for users of the API
+	_ "image/png"
+	"log"
+
+	"github.com/go-text/typesetting/shaping"
+	_ "golang.org/x/image/tiff" // load image formats for users of the API
+
+	"github.com/go-text/typesetting/opentype/api"
+	"github.com/nfnt/resize"
+)
+
+func (r *Renderer) drawBitmap(g shaping.Glyph, bitmap api.GlyphBitmap, img draw.Image, x, y float32) (advance float32, err error) {
+	adv := float32(0)
+
+	switch bitmap.Format {
+	case api.BlackAndWhite:
+		log.Println("black and white - TODO")
+	case api.JPG, api.PNG, api.TIFF:
+		pix, _, err := image.Decode(bytes.NewReader(bitmap.Data))
+		if err != nil {
+			return 0, err
+		}
+
+		h := r.FontSize * r.PixScale
```

Comment:
Is it safe to resize to a square ? Probably in practice ?
Otherwise, you could perhaps use the pix config to preserve the ratio.

Reply:
Ah good catch, I had meant to fix this before PR

Reply:
I have looked at using `Glyph.Width` but it seems to be around 2000* the size of `Glyph.XAdvance` - what am I missing?

Reply:
> I have looked at using `Glyph.Width` but it seems to be around 2000* the size of `Glyph.XAdvance` - what am I missing?

For which font ? It is weird indeed, since generally speaking `Glyph.XAdvance` should be of the same order as `Glyph.Width`.

Reply:
I was using the SVG emoji font packaged in this PR - using a debugger I stepped through the test code (as I figured the code to advance bitmap vs svg was basically identical).

Reply:
Maybe that's the design-space width for the SVG? I'm not sure how else to explain it. Even if it was being multiplied by 64 as a result of our inputs to the shaping process, 229470/64=3585, which is still an absurd size.

Reply:
Interesting idea but it doesn't seem so - the viewport is 0x0-64x64

Reply:
I don't know if it helps, but this is the `ext` picked up in the `GlyphExtents` call when shaping the character:

<img width="330" alt="Screenshot 2023-06-15 at 14 58 53" src="https://github.com/go-text/render/assets/294436/3a4f0b6b-137a-4d96-8b53-b233229f52f5">

Reply:
I'm not sure why those values are wrong, but can you use the equivalent values in the input glyph? https://github.com/go-text/typesetting/blob/main/shaping/output.go#L13

It has its own bearing and extent fields. Or do those have the same weird values?

Reply:
> I don't know if it helps, but this is the `ext` picked up in the `GlyphExtents` call when shaping the character:
> 
> <img alt="Screenshot 2023-06-15 at 14 58 53" width="330" src="https://user-images.githubusercontent.com/294436/246134060-3a4f0b6b-137a-4d96-8b53-b233229f52f5.png">

Thank you, this is definitively wrong, either due to a corrupted font file or a bug (more likely a bug). I'll try to dig deeper as soon as possible.

Reply:
Thanks for this. If I find out any more info I will share. Will try to find other, similar, fonts to see how the numbers shape up.

Reply:
See https://github.com/go-text/typesetting-utils/pull/11

Reply:
Epic, that works great now thanks!